drivers/hwspinlock/hwspinlock_handlers.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023 Sequans Communications
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwspinlock.h>
+#include <zephyr/syscall_handler.h>
+
+static inline int z_vrfy_hwspinlock_trylock(const struct device *dev, uint32_t id)
```

Comment:
will this API ever be used in userspace context?

Reply:
Good question, I think that it will not be used in userspace.. Let's remove hwspinlock handlers if you're agree.

Reply:
I prefer to keep the support in userspace, what is your point @gmarull ?